### PR TITLE
[tmva][sofie] Fix Python C-API reference handling in Keras parser

### PR DESCRIPTION
**Pull request by Shlok-Saxena**
# This pull request
Fixes Python C-API reference handling issues in the SOFIE Keras parser (`tmva/sofie_parsers/src/RModelParser_Keras.cxx`) to avoid memory leaks during model parsing.

The Keras parser relies on Python C-API calls (dictionary lookups, attribute access, string conversions). A few utilities and attribute extraction paths were creating temporary Python objects (new references) without releasing them, which could **lead to increasing memory usage** when parsing multiple models or repeatedly parsing within the same ROOT session.

This PR ensures temporary Python objects created during parsing are properly released, improving stability when parsing multiple models in the same session.

## Changes or fixes:
- Fixed memory leak in `GetValueFromDict()` by avoiding leaking temporary `PyUnicode` key objects created via `PyUnicode_FromString()`
- Simplified and corrected `PyStringAsString()` implementation by using `PyUnicode_AsUTF8()` for safe UTF-8 extraction (avoids leaking temporary objects and unsafe lifetime issues)
- Fixed Python reference leaks in `MakeKerasBatchNorm()` by properly releasing `PyObject*` returned from `PyObject_GetAttrString()` when retrieving BatchNorm weight names (`gamma`, `beta`, `moving_mean`, `moving_variance`)

## Checklist:

- [x] tested changes locally
- [ ] updated the docs (if necessary)

### CODE REVIEW

File: tmva/sofie_parsers/src/RModelParser_Keras.cxx [hunk 1]
```diff
@@ -63,7 +61,10 @@ std::vector<size_t> GetDataFromTuple(PyObject *tupleObject)
 
 PyObject *GetValueFromDict(PyObject *dict, const char *key)
 {
-   return PyDict_GetItemWithError(dict, PyUnicode_FromString(key));
+   PyObject *pyKey = PyUnicode_FromString(key);
+   PyObject *val = PyDict_GetItemWithError(dict, pyKey);
+   Py_DECREF(pyKey);
+   return val;
```
Comment (sanjibansg) [inline tmva/sofie_parsers/src/RModelParser_Keras.cxx:67]: Why do we need Py_DECREF here and the changed implementation?

File: tmva/sofie_parsers/src/RModelParser_Keras.cxx [hunk 2]
```diff
@@ -666,10 +667,18 @@ std::unique_ptr<ROperator> MakeKerasBatchNorm(PyObject* fLayer)
...
+      PyObject *meanName  = PyObject_GetAttrString(fMoving_Mean, "name");
+      PyObject *varName   = PyObject_GetAttrString(fMoving_Var, "name");
+      std::string fNScale = PyStringAsString(gammaName);
+      std::string fNB     = PyStringAsString(betaName);
+      std::string fNMean  = PyStringAsString(meanName);
+      std::string fNVar   = PyStringAsString(varName);
+      Py_DECREF(gammaName);
+      Py_DECREF(betaName);
+      Py_DECREF(meanName);
+      Py_DECREF(varName);
```
Comment (sanjibansg) [inline tmva/sofie_parsers/src/RModelParser_Keras.cxx:681]: same question on benefit of using this different implementation and Py_DECREF here.

**Comment by guitargeek** [comment]
Thanks for the PR!

Unfortunately the timing was not good, as the Keras parser has been re-implemented in native Python:

  * https://github.com/root-project/root/pull/20871

So there is no need anymore to use the Python C API in the Keras parser.
